webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import Select from '../../../../../common/forms/Select';
+import Checkbox from '../../../../../common/forms/Checkbox';
+import TextInput from '../../../../../common/forms/TextInput';
+import './disk.scss';
+
+const Disk = (
+  {
+    removeDisk,
+    updateDisk,
+    name,
+    storagePod,
+    datastores,
+    storagePods,
+    dataStore,
+    size,
+    thinProvision,
+    eagerZero
+  }
+) => {
+  const selectableStores = datastores.map(attribute => {
+    const key = Object.keys(attribute)[0];
+    const value = Object.values(attribute)[0];
+
+    return <option key={key} value={key}>{value}</option>;
+  });
+
+  const selectablePods = storagePods.map(attribute => {
+    const key = Object.keys(attribute)[0];
+    const value = Object.values(attribute)[0];
+
+    return <option key={key} value={key}>{value}</option>;
+  });
+
+  return (
+    <div className="disk-container">
+      <TextInput
+        value={name}
+        onChange={updateDisk.bind(this, 'name')}
+        label={__('Disk name')}
+      />
+
+      <Select
+        label={__('Storage Pod')}
+        value={storagePod}
+        onChange={updateDisk.bind(this, 'storagePod')}
+        options={selectablePods}
+      />
+
+      <Select
+        label={__('Data store')}
+        value={dataStore}
+        onChange={updateDisk.bind(this, 'dataStore')}
```

Comment:
@matanwerbner I think this is an issue as the react component is already initialized, it wont be mounted again.

afaiu steps to reproduce are:
1. have two different vmware compute resource
2. create a new host, select the first vmware, then change the compute resource to the second vmware

first vmware form will persist.